sumcheck/Cargo.toml

```diff
@@ -0,0 +1,31 @@
+[package]
+name = "winter-sumcheck"
+version = "0.1.0"
+description = "Implementation of the sum-check protocol for the LogUp-GKR protocol"
+authors = ["winterfell contributors"]
+readme = "README.md"
+license = "MIT"
+repository = "https://github.com/novifinancial/winterfell"
+documentation = "https://docs.rs/winter-sumcheck/0.1.0"
+categories = ["cryptography", "no-std"]
+keywords = ["crypto", "sumcheck", "iop"]
+edition = "2021"
+rust-version = "1.78"
+
+[features]
+concurrent = ["utils/concurrent", "dep:rayon", "std"]
+default = ["std"]
+std = ["utils/std"]
+
+[dependencies]
+air = { version = "0.9", path = "../air", package = "winter-air", default-features = false }
+crypto = { version = "0.9", path = "../crypto", package = "winter-crypto", default-features = false }
+math = { version = "0.9", path = "../math", package = "winter-math", default-features = false }
+utils = { version = "0.9", path = "../utils/core", package = "winter-utils", default-features = false }
+rayon = { version = "1.8", optional = true }
+smallvec = { version = "1.13", default-features = false }
+thiserror = { version = "1.0", git = "https://github.com/bitwalker/thiserror", branch = "no-std", default-features = false }
```

Comment:
Should we then default for the moment to not using it?

Reply:
I believe the `err_in_core` feature stabilizes in Rust 1.81, which will be [released on September 5th](https://releases.rs/docs/1.81.0/). We can (hopefully) expect `thiserror` to release a new version shortly after that.

I think it'd be better to use it, instead of having to rewrite all the errors twice. In the meantime, we could merge everything to a `logup-gkr` branch, which we only merge in `next` once `1.81` is out?